polynote-kernel/src/main/scala/polynote/kernel/PolyKernel.scala

```diff
@@ -29,78 +31,109 @@ import scala.tools.nsc.interactive.Global
 
 class PolyKernel private[kernel] (
   private val getNotebook: () => IO[Notebook],
-  val globalInfo: GlobalInfo,
+  val kernelContext: KernelContext,
   val outputDir: AbstractFile,
   dependencies: Map[String, List[(String, File)]],
   val statusUpdates: Publish[IO, KernelStatusUpdate],
-  subKernels: Map[String, LanguageKernel.Factory[IO]] = Map.empty
-) extends Kernel[IO] {
+  subKernels: Map[String, LanguageInterpreter.Factory[IO]] = Map.empty
+) extends KernelAPI[IO] {
 
   protected val logger: Logger = getLogger
 
   protected implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.fromExecutorService(Executors.newCachedThreadPool()))
 
-  private val launchingKernel = Semaphore[IO](1).unsafeRunSync()
-  private val kernels = new ConcurrentHashMap[String, LanguageKernel[IO]]()
+  private val launchingInterpreter = Semaphore[IO](1).unsafeRunSync()
+  private val interpreters = new ConcurrentHashMap[String, LanguageInterpreter[IO]]()
```

Comment:
Hmm, since we use `ConcurrentHashMap` a lot I wonder whether having a little utility wrapper could be worthwhile. 

For example we could give it a `get(String): Option` method so we don't have to keep writing `Option(myMap.get(key))` (and I think there might be a couple places where we forget to guard our `get`s like that too which could end up biting us.

Reply:
(just a thought, not for this PR)

Reply:
👍 it would be great to have a Scala wrapper for `ConcurrentHashMap`... but also sounds like a lot of work to do it right

Reply:
Hmm, actually - any reason not to use the normal `asScala` converter? Looks like there's a `JConcurrentMapWrapper` that gets used. 

Strangely I feel like I often see people use ConcurrentHashMap directly without converting but I'm not sure why...